internal/services/dispatcher/dispatcher.go

```diff
@@ -168,11 +168,13 @@ func (d *DispatcherImpl) Start(ctx context.Context) error {
 
 			return err
 		case task := <-taskChan:
-			err = d.handleTask(ctx, task)
+			go func(task *taskqueue.Task) {
```

Comment:
Task queue consumers should process tasks in a separate goroutine so they don't block.